### ISSUE TITLE
Plug & Play service sample throws when receiving command response

### ISSUE THREAD

**Issue by wickste** [Azure/azure-iot-sdk-java#658]
- **OS and version used:** Windows 10

- **Java runtime used:** 13.0.1

- **SDK version used:** latest preview with PnP support

# Description of the issue:
Plug & Play service sample throws when receiving command response

# Code sample exhibiting the issue:
>Tutorial: https://docs.microsoft.com/en-us/azure/iot-pnp/quickstart-connect-pnp-device-solution-java
>Code sample: https://github.com/Azure-Samples/azure-iot-samples-java/tree/master/digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample

When following the tutorial you will hit the exception on the step "Invoke a command"

# Console log of the issue:
```
C:\temp\pnp\azure-iot-samples-java\digital-twin\Samples\service\JdkSample>java -jar invoke-digital-twin-command/target/invoke-digital-twin-command-with-deps.jar
Invoking blink on device pnpDeviceJava with interface instance name environmentalSensor
Command invoked on the device successfully, the returned status was 200 and the request id was c89ead8d-0ab1-4081-ba2e-0e49516c1d0e
The returned PAYLOAD was
Exception in thread "main" com.fasterxml.jackson.core.JsonParseException: Unexpected character ('d' (code 100)): was expecting double-quote to start field name
 at [Source: (String)"{description=EnvironmentalSensor is blinking every 10 seconds.}"; line: 1, column: 3]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1804)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:693)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:591)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddName(ReaderBasedJsonParser.java:1767)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:692)
        at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer$Vanilla.deserialize(UntypedObjectDeserializer.java:648)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2992)
        at com.microsoft.azure.sdk.iot.digitaltwin.sample.InvokeDigitalTwinCommandSample.toPrettyFormat(InvokeDigitalTwinCommandSample.java:74)
        at com.microsoft.azure.sdk.iot.digitaltwin.sample.InvokeDigitalTwinCommandSample.main(InvokeDigitalTwinCommandSample.java:49)
```

**Comment by davilu** [comment]
@wickste  The issue is in device client side. Were you using sample code here? The payload didn't as JSON. Are you using device client sample code here? https://github.com/Azure-Samples/azure-iot-samples-java/blob/master/digital-twin/Samples/device/JdkSample/environmental-sensor-sample/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample/EnvironmentalSensor.java

**Comment by wickste** [comment]
@davilu Yes, I am using the device client sample. I am just following the quickstart tutorial from the official documentation.

**Comment by davilu** [comment]
@wickste Seems function `toPrettyFormat` is incorrect in https://github.com/Azure-Samples/azure-iot-samples-java/blob/master/digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample/InvokeDigitalTwinCommandSample.java.

**Comment by timtay-microsoft** [comment]
The command response payload won't necessarily be in json format, so the simplest fix here is to just not use pretty print on that returned payload in our digital twin command samples. I went ahead and made a PR on the samples repo to make this fix.

https://github.com/Azure-Samples/azure-iot-samples-java/pull/13

**Comment by timtay-microsoft** [comment]
I actually suspect the java digital twin service client is deserializing this response payload before it gets up to the sample code. Both the IoT explorer and C# digital twin service client sample can be used to test invoking a command against the java digital twin device client sample, and they see proper json like the below:

![image](https://user-images.githubusercontent.com/28789848/72099414-6ad52d80-32d5-11ea-8a21-616959c9badc.png)

![image](https://user-images.githubusercontent.com/28789848/72099467-82acb180-32d5-11ea-9422-fb5c84cc15c0.png)

Whereas the returned payload in the digital twin service client is not proper json as it is missing double quotes

![image](https://user-images.githubusercontent.com/28789848/72099511-9e17bc80-32d5-11ea-99f7-9f2ddccbb866.png)

**Comment by timtay-microsoft** [comment]
The fix for this should be checked in soon (#673)

**Comment by timtay-microsoft** [comment]
Seeing as how we did a complete refresh of plug and play, and deprecated the libraries used by this sample, I'm closing this issue as "Won't fix"
